tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,795 @@
+                }
+                descriptor.setOutputRealmLocation(Paths.get(outputRealmLocationOption).toString());
+            }
+            if (inputRealmLocationOption == null) {
+                errorHandler(ElytronToolMessages.msg.inputLocationNotSpecified());
+            } else {
+                Path inputPath = Paths.get(inputRealmLocationOption);
+                descriptor.setInputRealmLocation(inputPath.toString());
+                if(Files.notExists(inputPath)) {
+                    errorHandler(ElytronToolMessages.msg.inputLocationDoesNotExist());
+                }
+            }
+            if (hashEncodingOption == null) {
+                descriptor.setHashEncoding(Encoding.BASE64);
+            } else {
+                try {
+                    descriptor.setHashEncoding(Encoding.valueOf(hashEncodingOption.toUpperCase()));
+                } catch (IllegalArgumentException | NullPointerException e) {
+                    errorHandler(e);
+                }
+            }
+            if (populateOption == null) {
+                descriptor.setPopulate(true);
+            } else {
+                descriptor.setPopulate(Boolean.valueOf(populateOption));
+            }
+
+            if (levelsOption == null) {
+                descriptor.setLevels(2);
+            } else {
+                try {
+                    descriptor.setLevels(Integer.parseInt(levelsOption));
+                } catch (NumberFormatException e) {
+                    errorHandler(e);
+                }
+            }
+            if (encodedOption == null) {
+                descriptor.setEncoded(true);
+            } else {
+                try {
+                    descriptor.setEncoded(Boolean.parseBoolean(encodedOption));
+                } catch (IllegalArgumentException e) {
+                    errorHandler(e);
+                }
+            }
+            if (createCredentialStore != null) {
+                descriptor.setCreateCredentialStore(Boolean.valueOf(createCredentialStore));
+            } else {
+                descriptor.setCreateCredentialStore(true);
+            }
+
+            if (credentialStoreOption != null) {
+                descriptor.setCredentialStore(credentialStoreOption);
+            } else {
+                errorHandler(ElytronToolMessages.msg.credentialStoreDoesNotExist());
+            }
+
+            if (secretKeyAliasOption != null) {
+                descriptor.setSecretKeyAlias(secretKeyAliasOption);
+            } else {
+                descriptor.setSecretKeyAlias("key");
+            }
+            descriptors.add(descriptor);
+            checkDescriptorFields(descriptor);
+        } else if (inputRealmLocationOption != null | outputRealmLocationOption != null |
+                realmNameOption != null | credentialStoreOption != null | createCredentialStore != null |
+                secretKeyAliasOption != null | hashEncodingOption != null | levelsOption != null | populateOption != null) {
```

Comment:
I think encodedOption is the last one needed here.

Reply:
Added